blog/feeds.py

```diff
@@ -48,13 +62,22 @@ def feed_url(self):
     def feed_guid(self):
         return self.feed_url()
 
+    def feed_extra_kwargs(self, obj):
+        return {
+            'page': self.page,
+            'last_page': self.last_page,
+        }
+
     def items(self):
         posts = self.blog_index_page.get_posts()
 
-        if self.blog_index_page.feed_limit != 0:
-            return posts[:self.blog_index_page.feed_limit]
+        starting_index = ((self.page - 1) * self.feed_per_page)
+        last_index = (self.page * self.feed_per_page)
+
+        if self.blog_index_page.feed_limit != 0 and self.blog_index_page.feed_limit < last_index:
+            return posts[starting_index:self.blog_index_page.feed_limit]
 
-        return posts
+        return posts[starting_index:last_index]
```

Comment:
I'm looking at this code and the code in the `get_object` function above and I'm wondering if we're not re-inventing the wheel a little. What would you say about taking advantage of [Django's built-in pagination features](https://docs.djangoproject.com/en/2.2/topics/pagination/)? It seems to me like a lot of similar functionality is exposed to what is going on here, meaning we might be able to get away with fewer manual calculations.

Reply:
Yes. I could have used the in-built paginator check there. But I also had to check for the feed_limit logic, so I had to calculate anyways to check for that, so I just used the calculated results. I can definitely still use the in-built paginator. Let me know what you think.

Reply:
I was thinking that Django's paginator could be applied after the feed limit is applied, something like

```python
from django.core.paginator import Paginator

incidents = incident_index_page.get_incidents()[:feed_limit]  # plus the check for feed_limit == 0, and so on...
paginator = Paginator(incidents, incident_index_page.feed_per_page
```

And then the paginator gives you a lot of what we want here in terms of page calculations. But I haven't actually tried it so there might be something I'm not considering. Even though we're calculating the overall limit, I think having Django handle the page indices and related is still a fairly big gain.